test/parallel/test-process-config.js

```diff
@@ -20,26 +20,41 @@
 // USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 'use strict';
-require('../common');
+
+const common = require('../common');
+
+// Checks that the internal process.config is equivalent to the config.gypi file
+// created when we run configure.
+
 const assert = require('assert');
 const fs = require('fs');
 const path = require('path');
 
-// check for existence
+// Check for existence of `process.config`.
 assert(process.hasOwnProperty('config'));
 
-// ensure that `process.config` is an Object
+// Ensure that `process.config` is an Object.
 assert.strictEqual(Object(process.config), process.config);
 
 const configPath = path.resolve(__dirname, '..', '..', 'config.gypi');
+
+if (!fs.existsSync(configPath)) {
+  common.skip('config.gypi does not exist.');
+}
+
 let config = fs.readFileSync(configPath, 'utf8');
 
-// clean up comment at the first line
+// Clean up comment at the first line.
 config = config.split('\n').slice(1).join('\n').replace(/'/g, '"');
 config = JSON.parse(config, function(key, value) {
   if (value === 'true') return true;
   if (value === 'false') return false;
   return value;
 });
 
+// If the assert fails, it only shows 3 lines. We need all the output to
+// compare.
+console.log('config:', config);
+console.log('process.config:', process.config);
```

Comment:
Are these necessary?

Reply:
I believe as the comment states, in case the assertion in L60 fails the output it will show will be truncated. This logging is so that we'll have the raw input.
This could be minimized by wrapping the assert in a `try ... catch` and log in the catch before rethrowing

Reply:
See the comment. If the test passes nothing gets logged. If the test fails the assert just logs the first three lines, which is completely useless.

When this test fails it's usually because something is subtly different in your config, and you need to diff the two files to find out what it is. 

> This could be minimized by wrapping the assert in a try ... catch and log in the catch before rethrowing

Good idea.

Reply:
> If the test fails the assert just logs the first three lines, which is completely useless.

Sounds like a problem with `assert.deepStrictEqual`, but that should be addressed elsewhere.

Reply:
Added a commit to wrap in try/catch.

I think the better fix is to use `common.log()`, but we haven't implemented that yet. I'll raise a PR for it later.